bin/gcs_cli.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# Manipulate emulated GCS storage.
+
+# Usage: ./bin/gcs_cli.py CMD
+
+import os
+
+import click
+import requests
+import urllib3
+
+from google.auth.credentials import AnonymousCredentials
+from google.cloud import storage
+from google.cloud.exceptions import NotFound
+
+base_url = os.environ["STORAGE_EMULATOR_HOST"]
+
+
+def get_client():
+    # Create a session that is OK talking over insecure HTTPS
+    weak_https = requests.Session()
+    weak_https.verify = False
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+    return storage.Client(
+        credentials=AnonymousCredentials(), project="test", _http=weak_https
+    )
+
+
+@click.group()
+def gcs_group():
+    """Local dev environment GCS manipulation script"""
+
+
+@gcs_group.command("create")
+@click.argument("bucket_name")
+def create_bucket(bucket_name):
+    """Creates a bucket
+
+    Specify BUCKET name.
+
+    """
+    # https://github.com/fsouza/fake-gcs-server/blob/0c31d1573c14912fc58ae68118f9c9ece266756a/README.md?plain=1#L47
+    endpoint_url = f"{base_url}/storage/v1/b/{bucket_name}"
+
+    client = get_client()
+
+    try:
+        client.get_bucket(bucket_name)
+        # TODO Bucket already exists, but do we have access to it?
+        # s3 client has a head_bucket method which also checks access permissions:
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/head_bucket.html#head-bucket
+        # Address this for all uses of client.get_bucket in this script.
```

Comment:
This method isn't exactly equivalent to the S3 client's `head_bucket` method in that it doesn't check permissions. Given that this is an emulator, I don't think ACLs really apply anyway, so perhaps that's fine? That's what this TODO is about.

Reply:
I'm not entirely sure what you mean by this. If `get_bucket` verifies that the bucket was created, then that's good. If it doesn't do that, then we need to do something additional to make sure the bucket was created like list the contents of the bucket.

I don't think I'd worry about testing permissions the client has to the bucket.

Reply:
Okay thanks. Yes this `get_bucket` method will throw a `NotFound` error if the bucket doesn't exist, and only in that case do we call `create_bucket` in the `except` block downstream. [`create_bucket` will error](https://cloud.google.com/storage/docs/json_api/v1/buckets/insert) if it cannot create the bucket, or if the bucket already exists.